lib/api/apis.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+
+const utils = require('./utils');
+const database = require('../database');
+
+/*
+ * Determines Content Types and Content Type Keys
+ *
+ * @param {object} app - An Express app
+ *
+ * @returns {object} types - An object containing `key` of Content Type keys and `types` of all Content Types
+ */
+const types = app => {
+  let cts = _.cloneDeep(app.get('content-types')).map(ct => {
+    const type = {};
+
+    Object.keys(ct).forEach(attr => {
+      if (attr !== 'attributes') {
+        type[attr] = ct[attr];
+      }
+    });
+
+    return type;
+  });
+  const ctKeys = Object.keys(cts[0]).sort().filter(key => {
+    if (key === 'name') {
+      return false;
+    }
+
+    return true;
+  });
+  ctKeys.unshift('name');
+
+  cts = cts.map(ct => {
+    const type = ct;
+    type.meta = {
+      url: `/api/types/${ct.id}`,
+    };
+
+    return type;
+  });
+
+  return Promise.map(cts, ct => {
+    return database('live').count('id').where({
+      'type-slug': ct.id,
+    }).then(total => {
+      const type = ct;
+      type.meta.count = parseInt(total[0].count, 10);
+
+      return type;
+    });
+  }).then(contentTypes => {
+    return {
+      keys: ctKeys,
+      all: contentTypes,
+    };
+  });
+};
+
+/*
+ *  API for all Content
+ *
+ * @param {object} query - Request queries
+ *
+ * @returns {object} - All items and the pagination for the items
+ */
+// TODO: Tests
```

Comment:
nitpik: not true, right, tests exist, right? (they do, I checked); (will not keep writing this for other TODO: tests comments)